### ISSUE TITLE
Pint tests are failing

### ISSUE THREAD

**Issue by max-sixty** [pydata/xarray#8437]
### What is your issue?

Pint tests are failing on main — https://github.com/pydata/xarray/actions/runs/6817674274/job/18541677930

```
E               TypeError: no implementation found for 'numpy.min' on types that implement __array_function__: [<class 'pint.util.Quantity'>]
```

If we can't fix soon, should we disable?

CC @keewis

**Comment by TomNicholas** [comment]
> If we can't fix soon, should we disable?

What's the current feasibility of moving these tests out into pint-xarray? That's technically where they should live eventually imo

**Comment by keewis** [comment]
> What's the current feasibility of moving these tests out into pint-xarray?

should be possible, but running them in `xarray`'s CI has the advantage of testing at least one duck array, so we immediately get feedback if we're breaking something. Though if we replace them with more generic ones there's nothing that should stop us from moving the `pint` tests.

**Comment by dcherian** [comment]
we switched from numpy 1.24.X to 1.26 4 days ago, but stay at pint 0.20.1 when the latest is 0.22. Presumably some compatibility code for the __array_function__ codepath was removed.

**Comment by keewis** [comment]
For more details about this change, see https://github.com/pydata/xarray/issues/7420#issuecomment-1458106040 (I totally forgot I had fixed that). The first version to include that fix is indeed `0.22`, so we should use that.

**Comment by benbovy** [comment]
Would it be possible to have some opt-in for those tests?

It helped catching some regressions a couple of times during the index refactor so I think they are useful (although I agree more generic tests might be good if possible).

But in general I found that running those tests takes a significant part of running the whole xarray test suite.

**Comment by keewis** [comment]
the issue with opt-in is that they might not run at all, which means that any bugs they would have caught will slip by unnoticed. However, those tests being slow is a fair point, and I think this is something we can change: when writing these, speed was not one of my concerns.

**Comment by benbovy** [comment]
> the issue with opt-in is that they might not run at all, which means that any bugs they would have caught will slip by unnoticed

Yes they should be run at least once! Are they run in all, many or just a few items of the CI matrix? Would it be possible to opt-in for just one or a few matrix items if that's not the case already?

**Comment by keewis** [comment]
so far, every environment other than the `min-deps` contains `pint`, so that means almost every CI

**Comment by max-sixty** [comment]
> Would it be possible to opt-in for just one or a few matrix items if that's not the case already?

We could have one environment that runs `--run-nightly`, either opt-in, or limited to a single env, or only on merges (so we pick up corner-case failures quickly and can revert, without slowing down PR feedback)

**Comment by TomNicholas** [comment]
> so far, every environment other than the min-deps contains pint, so that means almost every CI

This seems like overkill, couldn't we just remove pint from a few of the envs? Depends which tests you want to run more quicky @benbovy ?
